### PR TITLE
[java] Fix the logic of generating TaskID in java

### PR DESCRIPTION
**Pull request by jovan-wong**
## What do these changes do?
Because the logic of generating `TaskID` in java is different from python's, there are many tests fail when we change the `Ray Core` code.
In this change,  I rewrote the logic of generating `TaskID` in java which is the same as the python's.

In java, we call the native method `_generateTaskId()` to generate a `TaskID` which is used in python.

### CODE REVIEW

File: java/runtime-common/src/main/java/org/ray/spi/LocalSchedulerLink.java [hunk 1]
```diff
@@ -21,5 +21,7 @@
 
   void notifyUnblocked();
 
+  UniqueID generateTaskId(UniqueID driverId, UniqueID parentTaskId, int taskIndex);
```
Comment (jovan-wong) [inline java/runtime-common/src/main/java/org/ray/spi/LocalSchedulerLink.java]: Should this be `genTaskId()` ?

File: java/runtime-dev/src/main/java/org/ray/spi/impl/MockLocalScheduler.java [hunk 2]
```diff
@@ -89,6 +89,11 @@ public void notifyUnblocked() {
 
   }
 
+  @Override
+  public UniqueID generateTaskId(UniqueID driverId, UniqueID parentTaskId, int taskIndex) {
+    throw new RuntimeException("Not implement here.");
```
Comment (raulchen) [inline java/runtime-dev/src/main/java/org/ray/spi/impl/MockLocalScheduler.java]: `implemented`

File: java/runtime-native/src/main/java/org/ray/spi/impl/DefaultLocalSchedulerClient.java [hunk 3]
```diff
@@ -113,6 +118,18 @@ public void reconstructObject(UniqueID objectId, boolean fetchOnly) {
   public void reconstructObjects(List<UniqueID> objectIds, boolean fetchOnly) {
     RayLog.core.info("reconstruct objects {}", objectIds);
     _reconstruct_objects(client, getIdBytes(objectIds), fetchOnly);
+    RayLog.core.info("task id is {}", UniqueIdHelper.taskIdFromObjectId(objectIds.get(0)));
+  }
+
+  @Override
+  public UniqueID generateTaskId(UniqueID driverId, UniqueID parentTaskId, int taskIndex) {
+    byte[] bytes = _generateTaskId(driverId.getBytes(), parentTaskId.getBytes(), taskIndex);
+    //TODO(qwang): This code will be removed once MultiReturns is removed.
```
Comment (raulchen) [inline java/runtime-native/src/main/java/org/ray/spi/impl/DefaultLocalSchedulerClient.java]: multi-return is already removed. update this?

File: java/runtime-common/src/main/java/org/ray/core/UniqueIdHelper.java [hunk 4]
```diff
@@ -3,274 +3,43 @@
...
-    } else {
-      long cid = rand.get().nextLong();
-      byte[] cbuffer = lbuffer.putLong(cid).array();
-      idBytes = MD5Digestor.digest(cbuffer, rand.get().nextLong());
-    }
-    setUniqueness(wbb, idBytes);
-    lbuffer.clear();
-    return taskId;
+    UniqueID retId = new UniqueID(taskId);
+    return retId;
```
Comment (raulchen) [inline java/runtime-common/src/main/java/org/ray/core/UniqueIdHelper.java]: just `return new ...`

File: java/runtime-common/src/main/java/org/ray/spi/LocalSchedulerProxy.java [hunk 5]
```diff
@@ -50,7 +50,7 @@ public RayObject submitActorCreationTask(UniqueID taskId, UniqueID createActorId
   private UniqueID[] genReturnIds(UniqueID taskId, int numReturns) {
     UniqueID[] ret = new UniqueID[numReturns];
     for (int i = 0; i < numReturns; i++) {
-      ret[i] = UniqueIdHelper.taskComputeReturnId(taskId, i, false);
+      ret[i] = UniqueIdHelper.taskComputeReturnId(taskId, i + 1);
```
[the remark this answers is not in the document]
Comment (jovan-wong) [reply java/runtime-common/src/main/java/org/ray/spi/LocalSchedulerProxy.java]: If we start it from 1, `ret[i]` will be changed to `ret[i - 1]`,
or we use a list as returning object.

**Pull request closed without merging**